### PR TITLE
Rework Chemistry-Reagents.dm

### LINKED ISSUE #3928: Polyacid damage inconsistency: deadly as smoke, very tame as reagent
Dump 200 units of polyacid on someone's unprotected face, they get less than 10 brute damage.
Feed them 20u of polyacid in a pill, they will veeeery slowly take 50 toxin damage. Even welder fluid is deadlier.
But put that polyacid in a spray bottle and puff 5 units onto the same person, they get whooping 30 brute and serious bleeding.
And jesus christ, polyacid smoke grenades near damn instantly fucking kill you if you stand anywhere near them.

### LINKED ISSUE #6294: Inaprovaline and Space Drugs Overdose Bug
Looking at reagents.dm, the on_mob_life() of inaprovaline and space drugs don't seem to call the parent function like all the other reagents with overdose thresholds do. I believe this would mean that it is not possible to overdose on them, despite overdose being intended as evidenced by the declaration of an overdose threshold for these two reagents.

### PR DESCRIPTION
**Pull request by ghost**
- All helper procs now handle sanity checks (holder, istype, etc). This massively reduces the amount of redundant checks for individual reagents, and limits shitcode
- Mob related procs now expect to be passed a mob/living, and will return if they don't
- Turf related procs now expect to be passed a turf/simulated, and will return if they don't
- All reagents should now call for parent. Fixes #6294 
- Removed all self.remove_reagent from reagent code (instead changed to custom_metabolism). Only exception is Discount's Dan Sauce
- Food and drink reagents now use FOOD_METABOLISM for both their custom_metabolism and nutrition_factor values
- All food reagents with non-null nutrition_factor now properly give nutrition. A few reagents have had this added (namely glycerol, hot sauce and cold sauce, and maybe another)
- Removed 0.01 metabolism from chemicals that didn't warrant it (aka ones that have zero effect on_mob_life like copper, sulfur, phosporus, etc). Reagents with a direct purpose like Spaceacilin weren't modified for balance reasons
- All reaction_turf procs that create cleanables (with the notable exception of blood) will now return if there already is a cleanable of the same type on the turf
- Moved reagent Destroy() back to the top of the file, instead of having it stuck right above vinegar
- A lethal amount of formatting

Potential balance changes : 
- Both water and holy water can now clean up turfs and mobs. Holy water now behaves mostly like water by damaging Greys, expanding monkey cubes and extinguishing fires so that the Chaplain doesn't accidentally sabotage firefighting supplies. You can't slip on holy water though
- Holy water now uses volume instead of hardcoded values. Should not change the damage output on vampires since those values were already calculated using a base formula (2 times regular, 0.4 mature vampire)
- Corn oil no longer behaves like water when sprayed on turfs (and certainly does not extinguish fires anymore)
- Unfucked acid (including water/holy water for Greys and Vampires). It now deals burn damage instead of brute damage, isn't capped on application (which forced the attacker to use rapid 5u sprays instead of a single beaker chug) and ingesting it properly multiplies the damage
- Spraying fuel on an object will now add fuel to the object's turf. Fixes #3928

There should be no other changes, please make sure to check because a lot of lines were modified

### CODE REVIEW

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -3715,7 +3702,7 @@
 			if(M.dna.mutantrace == "slime")
 				M.bodytemperature -= rand(15,20)
 	data++
-	holder.remove_reagent(src.id, FOOD_METABOLISM)
+	//holder.remove_reagent(src.id, FOOD_METABOLISM) //Handled by ..()
```
Comment (clusterfack) [inline code/modules/reagents/Chemistry-Reagents.dm]: Just remove the line totally

**Comment by clusterfack** [comment]
Standardize this file a little more, why is it checking if(holder) on some of the reagents and not on others. And remove the returns from the end of all these procs, oldcode central. And if you're going to comment out code that does nothing anymore just remove it.

**Comment by Iamgoofball** [comment]
ps. 0.01 metabolism rate is fucking atrocious on anything

makes 5u pills last forever, let alone 50u pills

**Comment by dylanstrategie** [comment]
I'll check to see what I can standardize, but I want to be EXTREMELY careful around Chemistry code, since if I break something it's going be be Breaking Bad

**Comment by SarahJohnson489** [comment]
Oh boy, remember to spray the gray away in cult mode. Can't deconvert them so into the gibber they go! Valids secured, let's go home lads. :-1:

**Comment by Iamgoofball** [comment]
You have changed enough lines that we can no longer view a diff of the file.

Nice.

**Comment by dylanstrategie** [comment]
Sarah, holy water is perfumed water. Of course it would seriously wound Greys

And you can deconvert them in other ways, like the Bible in a Chaplain's hands

**Comment by AutismSpider** [comment]
beat the grey over the head with the bible or rod to uncult.

**Comment by Exxion** [comment]
First few Cult rounds after this is merged and before people figure out that change will be fun for Grey players though
Nothing to be done about it though, as many changes fuck some shit up at least briefly

**Comment by dylanstrategie** [comment]
Will fix the conflicts once feature freeze is over, Cap'tn

**Comment by NigglyWiggly** [comment]
>Holy water kills greys

>Alternative is to be beaten into retardation by the chaplain

This is all just a plot to make me hate overzealous chaplains even more isn't it

**Comment by Probe1** [comment]
@AutismSpider I have to remind someone once a week that the rod doesn't deconvert anymore and hasn't for six months.  

And the scene is almost always they beat a man to death thinking they'd deconvert any time now.

**Comment by PJB3005** [comment]
Conflicting.

**Comment by dylanstrategie** [comment]
I'm aware. I will solve the conflicts once the freeze is over

**Comment by ShiggyPiggy** [comment]
:-1: solely for the metabolism rates change.

**Comment by dylanstrategie** [comment]
Only inert chemical's metabolism rates were changed. Reagents with an effect like spaceacilin weren't changed

**Comment by ShiggyPiggy** [comment]
You could technically still have up to 29.9 units of (n-1) chemicals required for a recipe as long as they don't react between themselves and inject the nth chemical when needed.
You might say that it's easier to keep a beaker with the needed chem on yourself, but what I'm talking about is, for example, Greyshit McMemes pumping himself full of anti-toxin, oxygen and sugar (I'm not sure if you can OD on either, but I'd prefer to be sure) and then, when needed, gobble gobble a beaker (ordinary, large, blue-space, doesn't matter) of carbon, and get a fuckton of bicaridine in his system. Bonus points for gobble-gobbling not more than 60 carbon per powergame moment (30 initial to make inaprov, 30 extra for 30u Bicaridine)

**Comment by ghost** [comment]
I will try and redo this more properly. Unfucking Chemistry_Reagents.dm is important

**Pull request closed without merging**